pallets/liquidity-pools/src/message.rs

```diff
@@ -367,562 +408,54 @@ where
-				tranche_id: decode::<16, _, _>(input)?,
-				investor: decode::<32, _, _>(input)?,
-				currency: decode_be_bytes::<16, _, _>(input)?,
-				tranche_tokens_payout: decode_be_bytes::<16, _, _>(input)?,
-				remaining_redeem_amount: decode_be_bytes::<16, _, _>(input)?,
-			}),
-			17 => Ok(Self::ExecutedCollectInvest {
-				pool_id: decode_be_bytes::<8, _, _>(input)?,
-				tranche_id: decode::<16, _, _>(input)?,
-				investor: decode::<32, _, _>(input)?,
-				currency: decode_be_bytes::<16, _, _>(input)?,
-				currency_payout: decode_be_bytes::<16, _, _>(input)?,
-				tranche_tokens_payout: decode_be_bytes::<16, _, _>(input)?,
-				remaining_invest_amount: decode_be_bytes::<16, _, _>(input)?,
-			}),
-			18 => Ok(Self::ExecutedCollectRedeem {
-				pool_id: decode_be_bytes::<8, _, _>(input)?,
-				tranche_id: decode::<16, _, _>(input)?,
-				investor: decode::<32, _, _>(input)?,
-				currency: decode_be_bytes::<16, _, _>(input)?,
-				currency_payout: decode_be_bytes::<16, _, _>(input)?,
-				tranche_tokens_payout: decode_be_bytes::<16, _, _>(input)?,
-				remaining_redeem_amount: decode_be_bytes::<16, _, _>(input)?,
-			}),
-			19 => Ok(Self::CancelInvestOrder {
-				pool_id: decode_be_bytes::<8, _, _>(input)?,
-				tranche_id: decode::<16, _, _>(input)?,
-				investor: decode::<32, _, _>(input)?,
-				currency: decode_be_bytes::<16, _, _>(input)?,
-			}),
-			20 => Ok(Self::CancelRedeemOrder {
-				pool_id: decode_be_bytes::<8, _, _>(input)?,
-				tranche_id: decode::<16, _, _>(input)?,
-				investor: decode::<32, _, _>(input)?,
-				currency: decode_be_bytes::<16, _, _>(input)?,
-			}),
-			21 => Ok(Self::ScheduleUpgrade {
-				contract: decode::<20, _, _>(input)?,
-			}),
-			22 => Ok(Self::CancelUpgrade {
-				contract: decode::<20, _, _>(input)?,
-			}),
-			23 => Ok(Self::UpdateTrancheTokenMetadata {
-				pool_id: decode_be_bytes::<8, _, _>(input)?,
-				tranche_id: decode::<16, _, _>(input)?,
-				token_name: decode::<TOKEN_NAME_SIZE, _, _>(input)?,
-				token_symbol: decode::<TOKEN_SYMBOL_SIZE, _, _>(input)?,
-			}),
-			24 => Ok(Self::DisallowInvestmentCurrency {
-				pool_id: decode_be_bytes::<8, _, _>(input)?,
-				currency: decode_be_bytes::<16, _, _>(input)?,
-			}),
-			_ => Err(parity_scale_codec::Error::from(
-				"Unsupported decoding for this Message variant",
-			)),
-		}
+		// Bincode serializes the enum tag variants as 4 bytes.
+		// But solidity side expect 1 byte.
+		// We emulate here as if it was 1
+		bytes[3..].into()
 	}
-}
 
-/// Decode a type that implements our custom [Codec] trait
-pub fn deserialize<const S: usize, O: Codec, I: Input>(
-	input: &mut I,
-) -> Result<O, parity_scale_codec::Error> {
-	let mut bytes = [0; S];
-	input.read(&mut bytes[..])?;
+	pub fn deserialize(data: &[u8]) -> Result<Self, DispatchError> {
+		// Bincode serializes the enum tag variants as 4 bytes.
+		// But solidity side expect 1 byte.
+		// We emulate here as if it was so.
+		let data = [vec![0, 0, 0], data.into()].concat();
```

Comment:
I do not like this, ideally we should overwrite the `serialize_enum` and `deserialize_enum` from `bincode` serializer to force 1 bytes per variant instead of 4 bytes.

Ref: https://docs.rs/bincode/latest/src/bincode/de/mod.rs.html#284

Reply:
Fixed with the custom serializer